src/index.js

```diff
@@ -163,20 +189,35 @@ class JanusAdapter {
       console.warn("localMediaStream not set. Will not publish audio or video");
     }
 
+    debug("pub waiting for offer");
     var offer = await peerConnection.createOffer();
-    await peerConnection.setLocalDescription(offer);
 
-    var answer = await handle.sendJsep(offer);
-    await peerConnection.setRemoteDescription(answer.jsep);
+    debug("pub waiting for ice and descriptions");
+    await Promise.all([
+      iceReady,
+      peerConnection.setLocalDescription(offer),
+      (async () => {
+        const answer = await handle.sendJsep(offer);
+        return peerConnection.setRemoteDescription(answer.jsep);
+      })(),
+    ]);
 
-    // Wait for the reliable datachannel to be open before we start sending messages on it.
-    await waitForEvent(reliableChannel, "open");
+    debug("pub waiting for webrtcup");
+    await this.getWebRtcUpPromise(handle.id).promise;
 
+    debug("pub waiting for join");
     // Send join message to janus. Listen for join/leave messages. Automatically subscribe to all users' WebRTC data.
     var message = await this.sendJoin(handle, this.room, this.userId, {notifications: true, data: true});
```

Comment:
This seems like it should be in flight while waiting for `webrtcup`.

Reply:
:+1: thanks